regex/hours.regex

```diff
@@ -1 +1 @@
-([0-9]{4}[AP]M)-([0-9]{4}[AP]M)
+([0-9]{4}[AP]M)?-([0-9]{4}[AP]M)?
```

Comment:
Why did you make each part of the time optional? This means that `-1200PM`, `0100PM-` and `-` are valid hours.